packages/markdown/processor/parse/magic-block-parser.js

```diff
@@ -57,40 +59,40 @@ function tokenize(eat, value) {
           {
             type: 'heading',
             depth: json.level || 2,
-            children: this.tokenizeInline(json.title, eat.now()),
+            children: 'title' in json ? this.tokenizeInline(json.title, eat.now()) : '',
           },
           json
         )
       );
     }
     case 'image': {
-      return eat(match)(
-        WrapPinnedBlocks(
-          json.images.map(img => {
-            const [url, title] = img.image;
-            return {
-              type: 'image',
-              url,
-              title,
-              alt: img.caption,
-              data: {
-                hProperties: {
-                  caption: img.caption,
-                },
-              },
-            };
-          })[0],
-          json
-        )
-      );
+      const imgs = json.images.map(img => {
+        const [url, title] = img.image;
+        return {
+          type: 'image',
+          url,
+          title,
+          alt: img.caption,
+          data: {
+            hProperties: {
+              caption: img.caption,
+            },
+          },
+        };
+      });
+      const img = imgs[0];
+
+      if (!img.url) return eat(match);
+      return eat(match)(WrapPinnedBlocks(img, json));
     }
     case 'callout': {
-      json.type = {
+      const types = {
         info: ['ℹ', 'info'],
         success: ['👍', 'okay'],
         warning: ['⚠️', 'warn'],
         danger: ['❗️', 'error'],
-      }[json.type];
+      };
+      json.type = json.type in types ? types[json.type] : [json.icon || '👍', json.type];
```

Comment:
Is this to handle cases where people have emoji in their docs that we don't recognize as types?

Reply:
Kind of, but it’s really more to account for cases in which users have hardcoded an `item` option in the magic block's JSON which doesn't correspond to one of our pre-approved themes. (Found this edge case in the [Branch docs](https://help.branch.io/).)